### PR TITLE
[TACHYON-1667] Improve efficiency of object store integration tests

### PR DESCRIPTION
**Pull request by calvinjia**
An initial proposal to address the slowness of object store tests. Because creating many small files is relatively expensive, we create all the working files for read tests once and reuse them. This improves the performance by a factor of how many test cases there are. 

A potential more permanent solution would be to write integration tests factoring in edge cases/the common case only and not enumerating an entire range of possibilities (for example, test less than 1 block file, 1 block file, 10 block file instead of 0.5, 1, 1.5 ... 10 blocks file)

### CODE REVIEW

File: tests/src/test/java/tachyon/client/BufferedBlockInStreamIntegrationTest.java [hunk 1]
```diff
@@ -59,6 +60,21 @@ public static final void beforeClass() throws Exception {
     sWriteBoth = StreamOptionUtils.getCreateFileOptionsCacheThrough(sTachyonConf);
     sWriteTachyon = StreamOptionUtils.getCreateFileOptionsMustCache(sTachyonConf);
     sWriteUnderStore = StreamOptionUtils.getCreateFileOptionsThrough(sTachyonConf);
+    sTestPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN; k <= MAX_LEN; k += DELTA) {
```
Comment (yupeng9) [inline tests/src/test/java/tachyon/client/BufferedBlockInStreamIntegrationTest.java]: What's this for? Could you add some comments?

File: tests/src/test/java/tachyon/client/FileInStreamIntegrationTest.java [hunk 2]
```diff
@@ -69,6 +70,21 @@ public static final void beforeClass() throws Exception {
     sWriteBoth = StreamOptionUtils.getCreateFileOptionsCacheThrough(sTachyonConf);
     sWriteTachyon = StreamOptionUtils.getCreateFileOptionsMustCache(sTachyonConf);
     sWriteUnderStore = StreamOptionUtils.getCreateFileOptionsThrough(sTachyonConf);
+    sTestPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN; k <= MAX_LEN; k += DELTA) {
```
Comment (yupeng9) [inline tests/src/test/java/tachyon/client/FileInStreamIntegrationTest.java]: Would it make sense to extract this into some util method?

File: tests/src/test/java/tachyon/client/LocalBlockInStreamIntegrationTest.java [hunk 3]
```diff
@@ -94,7 +107,6 @@ public void readTest1() throws IOException, TachyonException {
         Assert.assertEquals(cnt, k);
         Assert.assertTrue(BufferUtils.equalIncreasingByteArray(k, ret));
         is.close();
-        Assert.assertTrue(sFileSystem.getStatus(uri).getInMemoryPercentage() == 100);
```
Comment (gpang) [inline tests/src/test/java/tachyon/client/LocalBlockInStreamIntegrationTest.java]: why is the in-memory percentage not checked anymore? (here and and other test cases in this class)

**Comment by gpang** [comment]
I left a question in the code. Thanks!

File: tests/src/test/java/tachyon/client/LocalBlockInStreamIntegrationTest.java (hunk 3 above)
Comment (calvinjia) [inline tests/src/test/java/tachyon/client/LocalBlockInStreamIntegrationTest.java]: It doesn't make sense for it to be in memory here, there is no guarantee. The second check makes sense because we read with cache mode.

File: tests/src/test/java/tachyon/client/FileInStreamIntegrationTest.java (hunk 2 above)
Comment (calvinjia) [inline tests/src/test/java/tachyon/client/FileInStreamIntegrationTest.java]: Instead of a utils method it would be better to have some kind of inheritance for the stream tests, as they share a lot of common logic, but I think its a separate improvement from this solution.

File: tests/src/test/java/tachyon/client/BufferedBlockInStreamIntegrationTest.java (hunk 1 above)
Comment (calvinjia) [inline tests/src/test/java/tachyon/client/BufferedBlockInStreamIntegrationTest.java]: Updated, thanks.

**Comment by calvinjia** [comment]
@yupeng9 @gpang Comments addressed, please take a look

**Comment by yupeng9** [comment]
LGTM

**Comment by gpang** [comment]
LGTM
